app/controllers/idv/personal_key_controller.rb

```diff
@@ -38,6 +45,25 @@ def update
       redirect_to next_step
     end
 
+    def self.step_info
+      Idv::StepInfo.new(
+        key: :personal_key,
+        controller: self,
+        action: :show,
+        next_steps: [FlowPolicy::FINAL],
+        preconditions: ->(idv_session:, user:) do
+          return false unless idv_session.address_confirmed? || idv_session.phone_confirmed?
+
+          return false unless user.active_or_pending_profile
+
+          !idv_session.personal_key_acknowledged
+        end,
+        undo_step: ->(idv_session:, user:) {
+          idv_session.personal_key_acknowledged = nil
```

Comment:
Do we still set `idv_session.personal_key`? The code looked unclear to me. If we do, it should be set to nil here.

Reply:
`idv_session.personal_key` can be set as a side effect of profile minting (in `IdvSession::create_profile_from_applicant_with_password`). If it's not present, then PersonalKeyController regenerates it. I think the correct action is not to regenerate personal key here. Right now this is all kind of academic, since `confirm_idv_needed` will prevent the user from going back past the personal key controller

Reply:
`undo_step` is called from `clear_future_steps!` whenever someone submits a step. So if someone has a personal key in their session for some reason, I think it should be set to nil here. But as you say, it's somewhat academic now. We might want to have it in place as insurance against future changes.

Reply:
addressed in [e6cb841](https://github.com/18F/identity-idp/pull/9776/commits/e6cb8411cbd7ad13604460db283860ce95768c49)